src/vs/workbench/contrib/tasks/browser/abstractTaskService.ts

```diff
@@ -956,10 +953,14 @@ export abstract class AbstractTaskService extends Disposable implements ITaskSer
 	}
 
 	private _getFolderFromTaskKey(key: string): { folder: string | undefined; isWorkspaceFile: boolean | undefined } {
-		const keyValue: { folder: string | undefined; id: string | undefined } = JSON.parse(key);
-		return {
-			folder: keyValue.folder, isWorkspaceFile: keyValue.id?.endsWith(TaskSourceKind.WorkspaceFile)
-		};
+		try {
```

Comment:
for dependency tasks, somehow a string (the task map key) is getting set as the key for `_persistentTasks` so when it's parsed, throws an error since it's not an object. I've spent hours trying to figure out how that's set/stored and have no idea. This works around it

Reply:
I've logged everywhere I `store` and can't find where that's happening - it occurs after the correct setting/storing of  (recently used key, task)